src/display/api.js

```diff
@@ -1295,8 +1305,10 @@ var PDFWorker = (function PDFWorkerClosure() {
     },
 
     _setupFakeWorker: function PDFWorker_setupFakeWorker() {
-      warn('Setting up fake worker.');
-      globalScope.PDFJS.disableWorker = true;
+      if (!globalScope.PDFJS.disableWorker) {
+        warn('Setting up fake worker.');
```

Comment:
Making this warning appear only when PDF.js decides to fall back (and not when user explicitly sets the option).